py/torch_tensorrt/dynamo/conversion/impl/slice/ops.py

```diff
@@ -467,10 +469,40 @@ def tile(
         # prepend 1 to dims
         dims = (1,) * -diff + tuple(dims)
 
-    shapes = [i * j for i, j in zip(input.shape, dims)]
     starts = [0] * len(dims)
     strides = [1] * len(dims)
-    layer = ctx.net.add_slice(input, tuple(starts), tuple(shapes), tuple(strides))
+    if not (has_dynamic_shape_input) and not (has_dynamic_shape_dims):
+        shapes = [i * j for i, j in zip(input.shape, dims)]
+        layer = ctx.net.add_slice(input, tuple(starts), tuple(shapes), tuple(strides))
+    else:
+        shapes = None
+        index = 0
+        for i, j in zip(input.shape, dims):
+            if i == DYNAMIC_DIM:
+                i = get_shape(
+                    ctx, target, source_ir, name + f"_input_{index}", input, index
+                )
+            if j == DYNAMIC_DIM:
+                j = get_shape(
+                    ctx, target, source_ir, name + f"_dim_{index}", input, index
+                )
+            prod_shape = convert_binary_elementwise(
+                ctx,
+                target,
+                source_ir,
+                name + "_prod",
+                trt.ElementWiseOperation.PROD,
+                i,
+                j,
+            )
+            shapes.append(prod_shape)
+            layer = ctx.net.add_slice(
+                input, start=trt.Dims(), shape=trt.Dims(), stride=trt.Dims()
+            )
+            layer.set_input(1, tuple(starts))
+            layer.set_input(2, tuple(shapes))
```

Comment:
Looks like we are setting a list of ITensors as a second input. But shouldn't the second input be an ITensor (which is the concatenated output of these shapes ) ? or both of them work ?